### PR TITLE
PDR-4369 implement report sdk

### PR DESCRIPTION
**Pull request by mengjinglei**
## Changes
   
  - [ ] feature1
  - [ ] feature2 
  - [ ] fixbug1
  - [ ] fixbug2
 
## Reviewers

  - [ ] @[someone] please review
  - [ ] @[someotherone] please review
    
## Checklist
   
   - [ ] Rebased/mergable
   - [ ] Tests pass
   - [ ] CHANGELOG.md updated
   - [ ] Jira issue/task done

### CODE REVIEW

File: base/reqerr/error.go [hunk 1]
```diff
@@ -41,6 +41,13 @@ const (
 	ExecuteSqlError
 	EntityTooLargeError
 	InvalidDataSchemaError
+	ErrHeaderAppIdError
```
Comment (DangHexuan) [inline base/reqerr/error.go]: 这个是什么意思？咱们的报错还会涉及appId吗
还有下面的InvalidAppIdError

**Comment by gf53520** [comment]
LGTM

File: base/reqerr/error.go (hunk 1 above)
Comment (mengjinglei) [reply base/reqerr/error.go]: 已删

File: base/reqerr/error.go [hunk 2]
```diff
@@ -41,6 +41,12 @@ const (
 	ExecuteSqlError
 	EntityTooLargeError
 	InvalidDataSchemaError
+	ErrDBNameInvalidError
+	ErrInvalidSqlError
+	ErrInternalServerError
+	ErrInvalidAppIdError
```
Comment (DangHexuan) [inline base/reqerr/error.go]: 这个也要删掉吗

File: report/error.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	}
+	errId := msg[:errCodePrefixLen]
+
+	switch errId {
+	case "E8002":
+		err.ErrorType = reqerr.ErrDBNameInvalidError
+	case "E8003":
+		err.ErrorType = reqerr.ErrInvalidSqlError
+	case "E8004":
+		err.ErrorType = reqerr.ErrInvalidAppIdError
```
Comment (DangHexuan) [inline report/error.go]: 还有这里

File: base/reqerr/error.go (hunk 2 above)
Comment (mengjinglei) [reply base/reqerr/error.go]: fixed

File: report/error.go (hunk 3 above)
Comment (mengjinglei) [reply report/error.go]: fixed

**Comment by DangHexuan** [comment]
LGTM

**Pull request merged**